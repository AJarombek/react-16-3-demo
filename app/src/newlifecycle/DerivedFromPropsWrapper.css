/**
 * Stylesheet for the DerivedFromProps component.
 * Author: Andrew Jarombek
 * Date: 1/20/2020
 */

.derived-from-props-wrapper {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
}

.derived-from-props-wrapper > p {
    font-family: 'Sylexiad-Bold', Helvetica, sans-serif;
}

.derived-from-props-wrapper > p > code {
    color: black;
    background-color: transparent;
}
