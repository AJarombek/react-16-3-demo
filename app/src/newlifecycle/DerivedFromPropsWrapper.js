/**
 * Wrapper around the {@code getDerivedStateFromProps()} demo component.
 * @author Andrew Jarombek
 * @since 1/19/2020
 */

import React, { useState } from 'react';
import DerivedFromProps from './DerivedFromProps';
import { AJContainedButton } from 'jarombek-react-components';

const DerivedFromPropsWrapper = () => {
  const [show, setShow] = useState(false);

  return (
    <div className="derived-from-props-wrapper">
      <p><code>getDerivedStateFromProps()</code> Demonstration</p>
      <AJContainedButton onClick={() => setShow(!show)}>
        {show ? 'Hide Secret Message' : 'Show Secret Message'}
      </AJContainedButton>
      <DerivedFromProps show={show}/>
    </div>
  );
};

export default DerivedFromPropsWrapper;
