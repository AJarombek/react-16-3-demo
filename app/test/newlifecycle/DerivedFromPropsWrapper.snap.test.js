/**
 * Snapshot test for the {@link ../../src/newlifecycle/DerivedFromPropsWrapper} component.
 * @author Andrew Jarombek
 * @since 1/29/2020
 */

import React from 'react';
import DerivedFromPropsWrapper from '../../src/newlifecycle/DerivedFromPropsWrapper';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(<DerivedFromPropsWrapper />).toJSON();
  expect(tree).toMatchSnapshot();
});
