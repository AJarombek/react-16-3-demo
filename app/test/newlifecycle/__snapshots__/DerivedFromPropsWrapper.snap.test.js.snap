// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="derived-from-props-wrapper"
>
  <p>
    <code>
      getDerivedStateFromProps()
    </code>
     Demonstration
  </p>
  <div
    className="ajBaseButton-0-2-4 ajBaseButtonEnabled-0-2-5 aj-contained-button ajContainedButton-0-2-1 ajContainedButtonEnabled-0-2-2"
    onClick={[Function]}
  >
    <button
      disabled={false}
      type="button"
    >
      Show Secret Message
    </button>
  </div>
  <div
    className="derived-from-props"
  >
    <div
      className="secret-code-classified"
    >
      <p>
        CLASSIFIED
      </p>
    </div>
  </div>
</div>
`;
